### PR TITLE
REF: mix PeriodPseudoDtype into PeriodDtype

### PR DESCRIPTION
**Pull request by jbrockmendel**
This will allow us to start getting rid of `get_freq_code` usages and related libfrequencies functions that are unnecessarily roundabout.

### CODE REVIEW

File: pandas/core/dtypes/dtypes.py [hunk 1]
```diff
@@ -896,7 +896,8 @@ def __new__(cls, freq=None):
 
         elif freq is None:
             # empty constructor for pickle compat
-            u = object.__new__(cls)
+            # -10_000 corresponds to PeriodDtypeCode.UNDEFINED
```
Comment (jreback) [inline pandas/core/dtypes/dtypes.py:899]: is this actually hit in tests? what is the point of UNDEFINED? I mean, when does this become defined

File: pandas/core/dtypes/dtypes.py [hunk 2]
```diff
@@ -848,7 +848,7 @@ def __setstate__(self, state) -> None:
 
 
 @register_extension_dtype
-class PeriodDtype(PandasExtensionDtype):
+class PeriodDtype(dtypes.PeriodPseudoDtype, PandasExtensionDtype):
```
Comment (jreback) [inline pandas/core/dtypes/dtypes.py:851]: not a huge fan of PeriodPseuodDtype name

**Review by jreback — CHANGES_REQUESTED** [review]

File: pandas/core/dtypes/dtypes.py (hunk 2 above)
Comment (jbrockmendel) [reply pandas/core/dtypes/dtypes.py:851]: im open to suggestions.  joris objected to calling it PeriodDtype

File: pandas/core/dtypes/dtypes.py (hunk 1 above)
Comment (jbrockmendel) [reply pandas/core/dtypes/dtypes.py:899]: Yes.  We have a test that constructs `PeriodDtype()` which goes through this path.  the existing comment a couple lines up suggests this is for pickle compat

Comment (jreback) [reply pandas/core/dtypes/dtypes.py:899]: right, how does this get resolved though? do you see periods with undefined after this?

File: pandas/core/dtypes/dtypes.py (hunk 2 above)
Comment (jreback) [reply pandas/core/dtypes/dtypes.py:851]: this is basically a cython PeriodBaseDtype

File: pandas/core/dtypes/dtypes.py (hunk 1 above)
Comment (jbrockmendel) [reply pandas/core/dtypes/dtypes.py:899]: i think the idea is to use it as a placeholder until you can get a "real" PeriodDtype.  e.g. if you know that you are parsing a csv column to Period but dont yet know the freq.  just speculating here

File: pandas/core/dtypes/dtypes.py (hunk 2 above)
Comment (jbrockmendel) [reply pandas/core/dtypes/dtypes.py]: Updated with new name + green

**Review by jreback — APPROVED** [review]

**Comment by jreback** [comment]
looks good. slightly concerned about the UNDEFINED case (e.g. we don't assert this anywhere)........

**Comment by jbrockmendel** [comment]
> slightly concerned about the UNDEFINED case

It also shows up in period.pyx as FR_UND (xref #34588), so i dont think we'll be able to get rid of it, but we should at least be able to get to a point with _fewer_ of these enum-like codes lying around

**Pull request merged**